program/src/oracle/sort.c

```diff
@@ -0,0 +1,44 @@
+#include "sort.h"
+
+static inline void swap_int64( int64_t *a, int64_t *b )
+{
+  int64_t const temp = *a;
+  *a = *b;
+  *b = temp;
+}
+
+static inline int partition_int64( int64_t* v, int i, int j )
+{
+  int const p = i;
+  int64_t const pv = v[ p ];
+
+  while ( i < j ) {
+    while ( v[ i ] <= pv ) {
+      ++i;
+    }
+    while ( v[ j ] > pv ) {
+      --j;
+    }
```

Comment:
Are these guaranteed not to slide past `[0,size)`?

Reply:
the first loop is not, added a condition
the second loop is, because `pv = v[ i ]`